### ISSUE TITLE
Standardize flag case

### ISSUE THREAD

**Issue by xginn8** [balena-io/balena-cli#1700]
```
❯ balena version
11.30.15
```

In different commands and subcommands, camelCase is sometimes used (and case sensitive) and dash-case is used in other situations. One example from `balena build`:

```
    --projectName, -n <projectName>            Specify an alternate project name; default is the directory name                                                                                  
    --registry-secrets, -R <secrets.yml|.json> Path to a YAML or JSON file with passwords for a private Docker registry  
```

Imho dash case is much more typical for CLI programs.

**Comment by srlowe** [comment]
I guess if we do this it would need to happen at the next major release where we can introduce breaking changes.  Note that there are also commands (at least one) that have case sensitive flags, but not camelcase.  E.g.:
```
balena logs
-- service, -s
-- system, -S
```

**Comment by xginn8** [comment]
From my perspective, case sensitivity is fine and pretty standard in short flags (`-`) but strange in long flags (`--`). All long flags being dash case would be a massive improvement, I think.

**Comment by pdcastro** [comment]
Past CLI check-ins (including product team input) had also agreed that the ideal would be for all CLI commands to use dash case (lower case with dashes) for long flags. And yes, short flags can be case sensitive. Rather than a sudden breaking change in a major release, iirc there was consensus that the ideal would be to accept both the legacy style and dash case, with the help output and documentation using the dash case only. @srlowe, I suppose if we were ever to do that, the current oclif conversion would be a good opportunity... Sadly I suspect that the code to support this will not be very pretty.

**Comment by srlowe** [comment]
@pdcastro Yeah probably a good opportunity during the migration.  However I'll need to take a look at how to do it cleanly at the framework/command definition level (which will require some extension to default oclif behaviour), rather than having to check multiple flags for contents in the command implementation.

**Comment by pdcastro** [comment]
Yes, if oclif had a "flag aliases" concept (like Capitano does, I think), then the code would look all right at the command implementation level. Maybe the oclif parser could be made to look for flag aliases early on, and translate all aliases to the preferred style before proceeding with parsing. I guess there would be an "array of flags" at some early stage of parsing, in which case maybe an iteration over the array could replace any occurrences of "legacy flags" with the preferred style. 

(Hmm this might also mean that error messages would mention the preferred style instead of what the user actually typed, but that's OK if the help output was consistent -- it might even help nudging the user towards adopting the preferred style.)
